### PR TITLE
Cofactor swapping

### PR DESCRIPTION
**Pull request by AlbaLopez**

### CODE REVIEW

File: src/app/jobs/components/job-detail/job-detail.component.ts [hunk 1]
```diff
@@ -53,18 +57,21 @@ export class JobDetailComponent implements OnInit, OnDestroy {
...
             this.ninjaService.getPredict(jobId).subscribe((jobPrediction: PathwayResponse) => {
-              const jobs = JSON.parse(localStorage.getItem('jobs'));
-              const jobIndex = jobs.findIndex(((job) => job.id === parseInt(jobId, 10)));
-              jobs[jobIndex].state = jobPrediction.status;
+              jobPrediction.type = 'Pathway prediction';
+              this.model = this.store.pipe(
+                select(getModelName(jobPrediction.model_id)));
+              this.organism = this.store.pipe(
+                select(getOrganismName(jobPrediction.organism_id)));
+              console.log('JOB PREDICTION', jobPrediction);
```
Comment (kvikshaug) [inline src/app/jobs/components/job-detail/job-detail.component.ts]: Please remove

File: src/app/jobs/components/job-detail/job-detail.component.ts [hunk 2]
```diff
@@ -53,18 +57,21 @@ export class JobDetailComponent implements OnInit, OnDestroy {
...
+
               if (jobPrediction.result) {
+                if (jobPrediction.result.table[0].method === 'PathwayPredictor+CofactorSwap') {
```
Comment (kvikshaug) [inline src/app/jobs/components/job-detail/job-detail.component.ts]: * This string comparison looks a bit brittle, are you sure this is the best approach?
* This assumes that the method for the first row is always the same for all the results in the job, is that guaranteed? If so, it should probably be included outsided of the table (e.g. `jobPrediction.result.method`)

File: src/app/app-interactive-map/escherSettings.ts [hunk 3]
```diff
@@ -34,4 +34,5 @@ export default {
   tooltip: 'custom',
   tooltip_callbacks: null,
   enable_editing: false,
+  zoom_extent_canvas: true,
```
Comment (kvikshaug) [inline src/app/app-interactive-map/escherSettings.ts]: Is this related to the cofactor PR somehow?

File: src/app/jobs/components/job-detail/components/job-result-table-cofactor/job-results-table-cofactor.component.html [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    <td class="linkList">
+      <div *ngFor="let hp of element.heterologous_reactions">
+        <a
+           [hidden]="!reactions[hp]?.annotation.EC"
+           [attr.href]="ecLink(reactions[hp]?.annotation.EC)"
+           target="_blank">
+            {{reactions[hp]?.annotation.EC}}
+        </a>
+        <a
+          [hidden]="reactions[hp]?.annotation.EC || hp.startsWith('DM')"
```
Comment (kvikshaug) [inline src/app/jobs/components/job-detail/components/job-result-table-cofactor/job-results-table-cofactor.component.html]: It's not obvious to me why these reactions are hidden, could you maybe describe why in a comment?

File: src/app/jobs/components/job-detail/job-detail.component.ts [hunk 5]
```diff
@@ -53,18 +57,21 @@ export class JobDetailComponent implements OnInit, OnDestroy {
           selectNotNull(getJob, {jobId}),
           map((j) => {
             this.ninjaService.getPredict(jobId).subscribe((jobPrediction: PathwayResponse) => {
-              const jobs = JSON.parse(localStorage.getItem('jobs'));
-              const jobIndex = jobs.findIndex(((job) => job.id === parseInt(jobId, 10)));
-              jobs[jobIndex].state = jobPrediction.status;
+              jobPrediction.type = 'Pathway prediction';
```
Comment (kvikshaug) [inline src/app/jobs/components/job-detail/job-detail.component.ts]: It seems odd that the type declaration needs to be defined in the component. Isn't it the responsibility of the service to set this?

**Review by kvikshaug — COMMENTED** [review]
The filtering works nicely! But is it really necessary to copy the entire results table component for the cofactor changes? Seems the whole template and scss is almost exact duplicates.

File: src/app/jobs/components/job-detail/job-detail.component.ts (hunk 2 above)
Comment (AlbaLopez) [reply src/app/jobs/components/job-detail/job-detail.component.ts]: It it guaranteed that all the rows have the same method, but yes, I think maybe it is better included in the response.

File: src/app/jobs/components/job-detail/components/job-result-table-cofactor/job-results-table-cofactor.component.html (hunk 4 above)
Comment (AlbaLopez) [reply src/app/jobs/components/job-detail/components/job-result-table-cofactor/job-results-table-cofactor.component.html]: Yess, sometime we show the EC number, but sometimes the reactions doesnt have the EC number and we have to show the hp.

File: src/app/jobs/components/job-detail/job-detail.component.ts (hunk 5 above)
Comment (AlbaLopez) [reply src/app/jobs/components/job-detail/job-detail.component.ts]: It should be, but right now I dont have it in the service and I need it, maybe I can create cards, and work in the backend a little bit, what do you think? And I can include the method and the type :)

File: src/app/app-interactive-map/escherSettings.ts (hunk 3 above)
Comment (AlbaLopez) [reply src/app/app-interactive-map/escherSettings.ts]: Now I am not sure about this, so let me check it

**Review by kvikshaug — APPROVED** [review]
:+1:
